vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SnapshotManager.java

```diff
@@ -0,0 +1,233 @@
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Value("${page_size}")
+    private int pageSize;
+    
+    @Value("${resources_folder_path}")
+    private String resourcesPath;
+
+    @Autowired
+    @Qualifier("storageEngineDownloads")
+    private IStorageEngine storageEngineDownloads;
+
+    @Autowired
+    private IRenderingManager renderingManager;
+
+    private SequenceHistory sequenceHistory;
+
+    @Autowired
+    private ExhibitionSnapshotRepository exhibitionSnapshotRepository;
+
+    @Autowired
+    private SnapshotTaskRepository snapshotTaskRepository;
+    
+    @Autowired
+    private SpaceRepository spaceRepository;
+    
+    private final String RESOURCES_FOLDER_NAME = "resources";
+
+    /**
+     * Triggers the creation of an exhibition snapshot.
+     *
+     * This method initializes and saves an {@link ExhibitionSnapshot} and a {@link SnapshotTask},
+     *
+     * @return the created {@link ExhibitionSnapshot}
+     * @throws IOException                          if an I/O error occurs during the snapshot creation process
+     * @throws InterruptedException                 if the snapshot creation process is interrupted
+     * @throws SnapshotCouldNotBeCreatedException   if the snapshot could not be created due to any other errors
+     */
+    @Override
+    @Transactional
+    public ExhibitionSnapshot triggerExhibitionSnapshotCreation() throws IOException, InterruptedException, SnapshotCouldNotBeCreatedException {
+        String exhibitionFolderName = getExhibitionFolderName();
+        ExhibitionSnapshot exhibitionSnapshot = exhibitionSnapshotRepository.findByFolderName(exhibitionFolderName);        
+        if(exhibitionSnapshot == null ) {
+            exhibitionSnapshot = new ExhibitionSnapshot();
+        }
+        createSnapshotFolder(exhibitionSnapshot, exhibitionFolderName);
+
+        SnapshotTask snapshotTask =  createSnapshotTask(exhibitionSnapshot);
+
+        exhibitionSnapshot.setSnapshotTask(snapshotTask); 
+        exhibitionSnapshotRepository.save(exhibitionSnapshot);
+
+        try {
+            createSnapshot(resourcesPath, exhibitionFolderName, sequenceHistory, exhibitionSnapshot);
+            storageEngineDownloads.generateZip(exhibitionFolderName);
+        } catch (IOException | InterruptedException | FileStorageException e) {
+            throw new SnapshotCouldNotBeCreatedException(e.getMessage(), e);
+        }
+        return exhibitionSnapshot;
+    }
+
+    /**
+     * Creates and saves snapshotTask object.
+     * 
+     * @param exhibitionSnapshot {@link ExhibitionSnapshot} object for which the task is to be created
+     * @return the created       {@link SnapshotTask}
+     */
+    private SnapshotTask createSnapshotTask(ExhibitionSnapshot exhibitionSnapshot) {
+        SnapshotTask snapshotTask = new SnapshotTask();  
+        snapshotTask.setExhibitionSnapshot(exhibitionSnapshot);        
+        snapshotTaskRepository.save(snapshotTask);
```

Comment:
you should return the return value of the save method, so that whoever calls `createSnapshotTask` gets an object with id.